src/staking.rs

```diff
@@ -1052,1512 +1053,1487 @@ mod test {
         StargateFailing,
     >;
 
-    fn mock_router() -> BasicRouter {
-        Router {
-            wasm: WasmKeeper::new(),
-            bank: BankKeeper::new(),
-            custom: FailingModule::new(),
-            staking: StakeKeeper::new(),
-            distribution: DistributionKeeper::new(),
-            ibc: IbcFailingModule::new(),
-            gov: GovFailingModule::new(),
-            stargate: StargateFailing,
+    /// Test environment that simplifies initialization of test cases.
+    struct TestEnv {
+        api: MockApi,
+        storage: MockStorage,
+        router: BasicRouter,
+        block: BlockInfo,
+        validator_addr_1: Addr,
+        validator_addr_2: Addr,
+        validator_addr_3: Addr,
+        delegator_addr_1: Addr,
+        delegator_addr_2: Addr,
+        user_addr_1: Addr,
+    }
+
+    impl TestEnv {
+        /// Returns preconfigured test environment.
+        fn new(validator1: (u64, u64, u64), validator2: (u64, u64, u64)) -> Self {
```

Comment:
Consider making it more explicit. Instead of passing the tuples create some `ValidatorParams` or `ValidatorConfig` struct

Reply:
This is a nice suggestion; otherwise, the values are a little bit enigmatic.